DeprecationWarning raised in CI

Currently (re-)running the CI throws an error, due to a deprecation warning when loading CHGNet (see also: https://github.com/materialsproject/pymatgen/issues/4243):

```
==================================== ERRORS ====================================
_______________ ERROR collecting tests/test_mlip_calculators.py ________________
tests/test_mlip_calculators.py:8: in <module>
    from chgnet.model.model import CHGNet
.venv/lib/python3.10/site-packages/chgnet/model/__init__.py:3: in <module>
    from chgnet.model.dynamics import CHGNetCalculator, MolecularDynamics, StructOptimizer
.venv/lib/python3.10/site-packages/chgnet/model/dynamics.py:24: in <module>
    from pymatgen.analysis.eos import BirchMurnaghan
.venv/lib/python3.10/site-packages/pymatgen/analysis/eos.py:23: in <module>
    from pymatgen.core.units import FloatWithUnit
.venv/lib/python3.10/site-packages/pymatgen/core/__init__.py:12: in <module>
    from pymatgen.core.composition import Composition
.venv/lib/python3.10/site-packages/pymatgen/core/composition.py:22: in <module>
    from pymatgen.core.periodic_table import DummySpecies, Element, ElementType, Species, get_el_sp
.venv/lib/python3.10/site-packages/pymatgen/core/periodic_table.py:63: in <module>
    class ElementBase(Enum):
.venv/lib/python3.10/site-packages/pymatgen/core/periodic_table.py:755: in ElementBase
    @deprecated(is_rare_earth, message="is_rare_earth is corrected to include Y and Sc.", deadline=(2025, 1, 1))
.venv/lib/python3.10/site-packages/monty/dev.py:157: in deprecated
    raise_deadline_warning()
.venv/lib/python3.10/site-packages/monty/dev.py:86: in raise_deadline_warning
    raise DeprecationWarning(
E   DeprecationWarning: This function should have been removed on 2025-01-01.

---------- coverage: platform linux, python 3.10.16-final-0 ----------
Coverage XML written to file coverage.xml

=========================== short test summary info ============================
ERROR tests/test_mlip_calculators.py - DeprecationWarning: This function should have been removed on 2025-01-01.
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 4.56s ===============================
```

Since the warning is generated by `raise`, it seems that `--disable-warnings` and other pytest filters don't suppress this, so I'm not sure if there's a simple workaround?

One option could be to alter he environment variables that are checked before the warning is raised  e.g. GITHUB_REPOSITORY/CI - see [CI/repository checks](https://github.com/materialsvirtuallab/monty/blob/master/src/monty/dev.py#L51), but that may have unexpected side effects, so I don't think it's a good solution.

Hopefully this can be removed from `pymatgen` relatively quickly, but unfortunately they only support Python 3.10+, so we may need to reconsider 3.9 support to make use of any fixes.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -27,7 +27,9 @@ jobs:
         env:
           # show timings of tests
           PYTEST_ADDOPTS: "--durations=0"
-        run: uv run pytest --run-extra-mlips --cov janus_core --cov-append .
+        run: |
+          unset CI
+          uv run pytest --run-extra-mlips --cov janus_core --cov-append .
 
       - name: Report coverage to Coveralls
         uses: coverallsapp/github-action@v2

